core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -5706,6 +5706,13 @@ object ZIO extends ZIOCompanionPlatformSpecific {
       // TODO: Dotty doesn't infer this properly: io.bracket[R1, E1](a => UIO(a.close()))(use)
       acquireReleaseWith(io)(a => ZIO.succeed(a.close()))(use)
 
+    /**
+     * Like `withFinalizer, add a finalizer from AutoClosable.
+     */
+    def withFinalizerAutoCloseable[R1 <: R & Scope](implicit
+      trace: ZTraceElement
+    ): ZIO[R1 with Scope, E, A] = ZIO.acquireReleaseExit(io)((a, _) => ZIO.succeed(a.close()))
```

Comment:
Since you aren't doing anything with the `Exit` value here you can simplify this to use `ZIO.acquireRelease`.